mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp

```diff
@@ -455,11 +504,31 @@ struct BlockwiseGemmAccelRewritePattern
                                  Value sourceBase, Value mn_i, Value MN,
                                  Value k_i, Value K, Value mnPerMfmaGroup,
                                  Value mnWaves, Type ldsBufferElemType,
-                                 Type dataType, Value ldsOrig, Value regDest) {
+                                 Type dataType, Value ldsOrig, Value regDest,
+                                 bool isKContiguousDim,
+                                 int64_t copyMNPerThread) {
       // Compute source offset
       Value sourceOffset = accelEmitterPtr->computeLdsSourceOffset(
           kb, k_i, mnb, mn_i, b, MN, loc, sourceBase, mnWaves, laneId);
 
+      // Since the LDS layou is `k0 x rotate(m) x kpack` we want to rotate
+      // the dimension `n` before reading from LDS. Since we don't use
+      // any transform here, this is done by hand with rem/div operators
+      if (isKContiguousDim) {
```

Comment:
as before, I think we should be doing this only if kVecLen > 1.

Reply:
Yeah I "think" I see why you want that, but I am not sure this is the right solution. Resolving this discussion, but leaving the other one open.